### PR TITLE
use codecov

### PR DESCRIPTION
**Pull request by kne42**
see #363

### CODE REVIEW

**Comment by jni** [comment]
Looks like you haven't properly installed the requirements in the cov environment?

https://cirrus-ci.com/task/5949232106700800

**Comment by kne42** [comment]
ughh i just wanted to generate coverage without running the tests

**Comment by jni** [comment]
@kne42 soooo, should we turn off ignoring all the tests now?

**Comment by kne42** [comment]
@jni could you explain more what you mean by this comment?

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ignore:
+  - napari/_version.py
+  - "**/tests/*.py"
```
Comment (jni) [inline codecov.yml]: I noticed our code coverage is approximately 0 and wondered if this might be the culprit.

Comment (jni) [reply codecov.yml]: We don't have a similar line in scikit-image: https://github.com/scikit-image/scikit-image/blob/master/.codecov.yml

**Comment by kne42** [comment]
this is ready for re-review @jni

**Review by sofroniewn — APPROVED** [review]
Looks good to me

File: .cirrus.yml [hunk 2]
```diff
@@ -92,3 +90,30 @@ task:
...
+    - apt-get update
+    - apt-get install -y libgl1-mesa-glx libqt5x11extras5
+
+  install_script:
+    - python -m pip install --retries 3 --upgrade pip
+    - pip install -r requirements.txt
+    - pip install pytest-cov codecov
+
+  coverage_script:
+    - pytest --cov=./napari --collect-only
```
Comment (jni) [inline .cirrus.yml]: @kne42 are you sure about the 'collect-only'?

Comment (kne42) [reply .cirrus.yml]: nope, I can try and we can see if that changes anything ;P

**Comment by jni** [comment]
Fun!

```
  File "/usr/local/bin/pytest", line 10 in <module>
/tmp/scriptsc67cf22746e995af5a25367951baa2ff.sh: line 5:  2054 Aborted                 (core dumped) pytest --cov=./napari
Exit status: 134
```

:man_facepalming:

**Comment by kne42** [comment]
@jni CI is passing again, wanna give the coverage another look?

**Comment by jni** [comment]
I was hoping for a new report, but I guess I'm happy to wait for the next PR. =)

**Pull request merged**